http/src/main/java/io/hyperfoil/http/config/HttpBuilder.java

```diff
@@ -98,6 +98,16 @@ public String host() {
       return host;
    }
 
+   public int port() {
+      if (port != -1) {
+         return port;
+      } else if (protocol != null) {
+         return protocol.portOrDefault(port);
+      } else {
+         return Protocol.fromPort(port).portOrDefault(port);
```

Comment:
I think that if `port == -1 && protocol == null` this method should rather throw a BenchmarkDefininitionException

Reply:
(because in this situation it can unexpectedly fall back to port 80)

Reply:
ok, done